src/pages/iou/request/step/IOURequestStepCurrency.js

```diff
@@ -78,8 +82,10 @@ function IOURequestStepCurrency({
      */
     const confirmCurrencySelection = (option) => {
         Keyboard.dismiss();
-        IOU.setMoneyRequestCurrency_temporaryForRefactor(transactionID, option.currencyCode);
-        navigateBack();
+        if (pageIndex !== 'confirm') {
+            IOU.setMoneyRequestCurrency_temporaryForRefactor(transactionID, option.currencyCode);
+        }
+        navigateBack(option.currencyCode);
```

Comment:
A minor UI issue here. As we navigate back without ensuring that the Onyx has indeed updated the new currency, the new currency does not immediately get displayed in the previous screen. This resulted in issue #41515